File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/ItemCatService.java
package com.pinyougou;

import java.util.List;

import com.github.pagehelper.PageInfo;
import com.pinyougou.pojo.ItemCat;

/**
 * 服务层接口
 *
 * @author Administrator
 */
public interface ItemCatService {

    /**
     * 返回全部列表
     *
     * @return
     */
    public List<ItemCat> findAll();


    /**
     * 返回分页列表
     *
     * @return
     */
    public PageInfo<ItemCat> findPage(int pageNum, int pageSize);


    /**
     * 增加
     */
    public void add(ItemCat itemCat);


    /**
     * 修改
     */
    public void update(ItemCat itemCat);


    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    public ItemCat findOne(Long id);


    /**
     * 批量删除
     *
     * @param ids
     */
    public void delete(Long[] ids);

//	/**
//	 * 分页
//	 *
//	 * @param pageNum  当前页 码
//	 * @param pageSize 每页记录数
//	 * @return
//	 */
//	public PageInfo<ItemCat> findPage(ItemCat itemCat, int pageNum, int pageSize);

    /**
     * 根据上级ID返回列表
     * @return
     */
    public List<ItemCat> findByParentId(Long parentId);
}

File: pinyougou_parent/pinyougou_shop_web/src/main/webapp/js/service/itemCatService.js
//服务层
app.service('itemCatService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('/itemCat/findAll.shtml');
	}
	//分页 
	this.findPage=function(page,size){
		return $http.get('/itemCat/findPage.shtml?page='+page+'&size='+size);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('/itemCat/findOne.shtml?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('/itemCat/add.shtml',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('/itemCat/update.shtml',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('/itemCat/delete.shtml?ids='+ids);
	}
	//搜索
	this.search=function(page,size,searchEntity){
		return $http.post('/itemCat/search.shtml?page='+page+"&size="+size, searchEntity);
	}

	//根据父id查询分类列表
	this.findByParentId=function(parentId){
		return $http.post('/itemCat/findByParentId.shtml?parentId='+parentId);
	}
});
